org.eclipse.gef.examples.ui.pde/src/org/eclipse/gef/examples/ui/pde/internal/wizards/ProjectUnzipperNewWizard.java

```diff
@@ -317,36 +303,33 @@ private void extractProject(File projectFolderFile, URL url, IProgressMonitor mo
 	 * @param projectFolderFile The folder where to unzip the project archive
 	 * @param monitor           Monitor to display progress and/or cancel operation
 	 * @throws IOException
-	 * @throws FileNotFoundException
 	 * @throws InterruptedException
 	 */
-	private void unzip(ZipFile zipFile, File projectFolderFile, IProgressMonitor monitor)
-			throws IOException, FileNotFoundException, InterruptedException {
+	private static void unzip(ZipFile zipFile, File projectFolderFile, IProgressMonitor monitor)
+			throws IOException, InterruptedException {
 
-		Enumeration e = zipFile.entries();
+		Enumeration<? extends ZipEntry> e = zipFile.entries();
 
 		while (e.hasMoreElements()) {
-			ZipEntry zipEntry = (ZipEntry) e.nextElement();
+			ZipEntry zipEntry = e.nextElement();
 			File file = new File(projectFolderFile, zipEntry.getName());
 
-			if (false == zipEntry.isDirectory()) {
+			if (!zipEntry.isDirectory()) {
 
 				/*
 				 * Copy files (and make sure parent directory exist)
 				 */
 				File parentFile = file.getParentFile();
-				if (null != parentFile && false == parentFile.exists()) {
+				if (null != parentFile && !parentFile.exists()) {
 					parentFile.mkdirs();
 				}
 
 				Path path = new Path(file.getPath());
 				if (path.getFileExtension().equals("java")) { //$NON-NLS-1$
-					InputStreamReader is = null;
-					OutputStreamWriter os = null;
-
-					try {
-						is = new InputStreamReader(zipFile.getInputStream(zipEntry), "ISO-8859-1"); //$NON-NLS-1$
-						os = new OutputStreamWriter(new FileOutputStream(file), ResourcesPlugin.getEncoding());
+					try (InputStreamReader is = new InputStreamReader(zipFile.getInputStream(zipEntry),
+							StandardCharsets.ISO_8859_1);
+							OutputStreamWriter os = new OutputStreamWriter(new FileOutputStream(file),
+									ResourcesPlugin.getEncoding());) {
```

Comment:
I think Windows-1252 is not really an appropriate encoding anymore. Given that this is an example, I would propose to use UTF-8 instead.

```suggestion
					try (InputStreamReader is = new InputStreamReader(zipFile.getInputStream(zipEntry),
							StandardCharsets.UTF_8);
							OutputStreamWriter os = new OutputStreamWriter(new FileOutputStream(file),
									ResourcesPlugin.getEncoding());) {
```

Reply:
you are totally right. Changed that accordingly.